### PR TITLE
Added some tests

### PR DESCRIPTION
**Pull request by iCarrrot**
I have created some index and details view tests.
I had to change save() function in models module to be able to create this tests.

### CODE REVIEW

File: izp/polls/tests.py [hunk 1]
```diff
@@ -1,3 +1,177 @@
...
+    given number of `days` offset to now (negative for questions published
+    in the past, positive for questions that have yet to be published).
+    """
+    if days != 0 and start == 0 and end == 0:
+        time = timezone.now() + datetime.timedelta(days=days)
+        return Question.objects.create(question_text=question_text, start_date=time)
+    if days != 0 and start != 0 and end == 0:
+        time = start + datetime.timedelta(days=days)
+        return Question.objects.create(question_text=question_text, start_date=start, end_date=time)
+    return Question.objects.create(question_text=question_text, start_date=start, end_date=start)
```
Comment (czapiga) [inline izp/polls/tests.py]: ```end_date=start```? I think you meant ```end_date=end```. How did tests pass? I think it should be impossible to create questions when ```start_date==end_date```. Why there is no test for that case? If it's actually possible then it is bug and you should prepare fix.

File: izp/polls/tests.py [hunk 2]
```diff
@@ -1,3 +1,177 @@
...
-# Create your tests here.
+
+def create_question(question_text, days=0, start=0, end=0):
+    """
+    Create a question with the given `question_text` and published the
+    given number of `days` offset to now (negative for questions published
+    in the past, positive for questions that have yet to be published).
+    """
+    if days != 0 and start == 0 and end == 0:
+        time = timezone.now() + datetime.timedelta(days=days)
```
Comment (czapiga) [inline izp/polls/tests.py]: I think you can override ```end``` variable instead of creating next one

File: izp/polls/models.py [hunk 3]
```diff
@@ -4,20 +4,22 @@
...
     question_text = models.CharField('Pytanie', max_length=200)
-    start_date = models.DateTimeField('Data rozpoczęcia', blank=True, default=timezone.now)
+    start_date = models.DateTimeField(
+        'Data rozpoczęcia', blank=True, default=timezone.now)
     end_date = models.DateTimeField('Data zakończenia', blank=True)
     time = models.IntegerField('Czas na odpowiedź [minuty]', default=5)
     access_codes = ['AAA', 'BBB', 'CCC']  # TODO generate random codes
 
-    def save(self):
+    def save(self, force_insert=False, force_update=False, using=None):
```
Comment (czapiga) [inline izp/polls/models.py]: What for? ```force_insert=False``` and ```force_update=False``` are default values so I think you can revert this change.
Why are you setting ```using=None```?

Comment (iCarrrot) [reply izp/polls/models.py]: Save() method is overwritten here, so default values have to be set here.

File: izp/polls/tests.py (hunk 2 above)
Comment (iCarrrot) [reply izp/polls/tests.py]: done

File: izp/polls/tests.py (hunk 1 above)
Comment (iCarrrot) [reply izp/polls/tests.py]: Well, this case wasn't use, but I've corrected this.

File: izp/polls/models.py (hunk 3 above)
Comment (czapiga) [reply izp/polls/models.py]: You're not even using them.

Comment (iCarrrot) [reply izp/polls/models.py]: I wanted to have this, to not have a problem in future with debugging.
Here is stack answer:
https://stackoverflow.com/questions/12760262/django-autoupdate-user-save-got-an-unexpected-keyword-argument-force-insert

**Comment by iCarrrot** [comment]
Done

File: izp/polls/tests.py [hunk 4]
```diff
@@ -1,3 +1,177 @@
...
+        future_response = self.client.get(url)
+        self.assertContains(future_response, "Głosowanie nie jest aktywne")
+
+        past_question = create_question(
+            question_text='Past question.', days=-5)
+        url = reverse('polls:detail', args=(past_question.id,))
+        past_response = self.client.get(url)
+        self.assertContains(past_response, "Głosowanie nie jest aktywne")
+
+        actual_question = create_question(
```
Comment (czapiga) [inline izp/polls/tests.py]: Well.. we've talked about it. Such test case should not pass. I've told you to check if creating questions with the same start and end date is possible. Please fix that.

File: izp/polls/tests.py [hunk 5]
```diff
@@ -1,3 +1,181 @@
...
+        create_question(question_text="Past question.", days=-30)
+        response = self.client.get(reverse('polls:index'))
+        self.assertQuerysetEqual(
+            response.context['questions_list'],
+            ['<Question: Past question.>']
+        )
+
+    def test_question_with_same_start_and_end_time(self):
+        """
+        Questions with a start_date which is equal to and_date should not be desplayd.
```
Comment (czapiga) [inline izp/polls/tests.py]: and_date -> end_date
desplayd -> displayed
There are plugins for most of IDEs which check spelling. Please install one.

File: izp/polls/tests.py [hunk 6]
```diff
@@ -1,3 +1,181 @@
...
+        index page.
+        """
+        create_question(question_text="Past question.", days=-30)
+        response = self.client.get(reverse('polls:index'))
+        self.assertQuerysetEqual(
+            response.context['questions_list'],
+            ['<Question: Past question.>']
+        )
+
+    def test_question_with_same_start_and_end_time(self):
```
Comment (czapiga) [inline izp/polls/tests.py]: This test case is invalid. I was so confused reading this. I've merged your code, run tests and it failed like it should. You haven't even checked that before updating PR.

File: .gitignore [hunk 7]
```diff
@@ -4,3 +4,4 @@
 db.sqlite3
 izp/polls/migrations/*.py
 izp/.vscode/*
```
Comment (czapiga) [inline .gitignore]: I think line 6 can be removed. If not please replace lines 6,7 with single pattern for .vscode folders

File: izp/polls/models.py [hunk 8]
```diff
@@ -1,26 +1,32 @@
+# -*- coding: utf-8 -*-
```
Comment (czapiga) [inline izp/polls/models.py]: Remove

File: izp/polls/tests.py [hunk 9]
```diff
@@ -1,8 +1,201 @@
+# -*- coding: utf-8 -*-
```
Comment (czapiga) [inline izp/polls/tests.py]: Remove
